pettingzoo/sisl/waterworld/waterworld_base.py

```diff
@@ -515,17 +598,26 @@ def move_objects(objects):
                 for obj in objects:
                     # Move objects
                     obj.set_position(obj.position + self.cycle_time * obj.velocity)
-                    # Bounce object if it hits a wall
-                    for i in range(len(obj.position)):
-                        if obj.position[i] >= 1 or obj.position[i] <= 0:
-                            obj.position[i] = np.clip(obj.position[i], 0, 1)
-                            obj.velocity[i] = -1 * obj.velocity[i]
+
+                    # TODO: vectorize this calculation the same way in the collision code
+                    distance = abs(obj.position - self.initial_obstacle_coord) + obj._radius
+                    if (distance[0] ** 2 + distance[1] ** 2 > 0.25):
```

Comment:
It's possible to use linalg norm here, then it's also possible to remove the 0.25 magic number